pymeasure/instruments/anaheimautomation/dpseriesmotorcontroller.py

```diff
@@ -0,0 +1,357 @@
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from time import sleep
+from enum import IntFlag
+from pymeasure.adapters import VISAAdapter
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range, truncated_range, strict_discrete_set
+
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class DPSeriesErrors(IntFlag):
+    """ IntFlag type to decode error register queries. Error codes are as follows:
+    
+        0: no error
+        1: Receive Overflow Error: serial communications had a receiving error.
+        2: Encoder Error 1: encoder needed to correct the motor position.
+        4: Encoder Error 2: encoder could not finish motor position correction.
+        8: Command Error: a bad command was sent to the controller.
+        16: Motor Error: motor speed profiles are set incorrectly.
+        32: Range Overflow Error: go to position has an overflow error.
+        64: Range Error: invalid number of commands and characters sent to the controller.
+        128: Transmit Error: Too many parameters sent back to the pc.
+        256: Mode Error: Controller is in a wrong mode.
+        512: Zero Parameters Error: Command sent to the controller that expected to see
+                                    parameters follow, but none were given.
+        1024: Busy Error: The controller is busy indexing (moving a motor).
+        2048: Memory Range Error: Specified address is out of range.
+        4096: Memory Command Error: Command pulled from memory is invalid.
+        8192: Thumbwheel Read Error: Error reading the thumbwheel, or thumbwheel is not present.
+    """
+    NO_ERR = 0
+    RCV_OVERFLOW_ERR = 1
+    ENC_ERR_1 = 2
+    ENC_ERR_2 = 4
+    CMD_ERR = 8
+    MOT_ERR = 16
+    RANGE_OVERFLOW_ERR = 32
+    RANGE_ERR = 64
+    TX_ERR = 128
+    MODE_ERR = 256
+    ZERO_PARAMS_ERR = 512
+    BUSY_ERR = 1024
+    MEM_RANGE_ERR = 2048
+    MEM_CMD_ERR = 4096
+    THBWHEEL_ERR = 8192
+
+    
+class DPSeriesMotorController(Instrument):
+    """Base class to interface with Anaheim Automation DP series stepper motor controllers. This driver has been tested
+       with the DPY50601 and DPE25601 motor controllers.
+    """
+
+    address = Instrument.control(
+        "%", "~%i",
+        """Integer property representing the address that the motor controller uses for serial communications.""",
+        validator=strict_range,
+        values=[0, 99],
+        get_process=lambda addr: int(addr),
```

Comment:
Did you see [this comment](https://github.com/samcondon4/pymeasure/commit/78eaee15d1a310eb53476c424dc23b894a484eb3#r60793080)?

> note: this is not well-documented at the moment, but if you pass `cast=int` here instead this does the same thing. (leftover kwargs get passed on to `adapter.values()`)

*btw, `int` is a callable just like a lambda, so there's no need to define a lambda here anyway.*

cf. #530